### ISSUE TITLE
MPC/MPFR libraries not found *_LIBRARIES=...

### ISSUE THREAD

**Issue by zerothi** [symengine/symengine#1576]
When trying to compile symengine with non-standard MPC/MPFR library directories one get this:

```bash
$> cmake -DMPFR_LIBRARIES=... -DMPFR_INCLUDE_DIRS=...

<lots of output>
  Could NOT find MPFR (missing: MPFR_LIBRARIES MPFR_INCLUDE_DIRS)
```
this error message is *very* annoying, since when looking in the FindMPFR.cmake files one finds that we need to do:
```bash
$> cmake -DMPFR_LIBRARY=... -DMPFR_INCLUDE_DIR=...
```
and then everything works.
The error message is at best misleading ;)

**Comment by certik** [comment]
Yes, this is annoying about CMake. I don't know how to best fix it. This is implemented here:
https://github.com/symengine/symengine/blob/4f7cd0e74901e4e6cabcfab2cdcbd72c85d455ef/cmake/FindMPFR.cmake#L6
For consistency, we use the `*_LIBRARIES` variable for all our 3rd party libraries.

I think one solution is to remove `MPFR_LIBRARIES` and just use the `MPFR_LIBRARY` variable. And also switch to the modern CMake way of handling dependencies, I started the work in #757, and in fact I just noticed that I did fix the MPFR problem you reported here:

https://github.com/symengine/symengine/pull/757/files#diff-8fb09aa8eb082f13924833423b4c283d

The work in #757 needs to be finished: most probably by splitting the task into many smaller tasks that can be merged one by one, so that it's not one huge PR.

**Comment by zerothi** [comment]
Regarding CMake, this is constantly bugging me about CMake files. From the error message you think that the variables are `*_LIBRARIES`, but once you look in the files you *have* to use the `*_LIBRARY` files because `*_LIBRARIES` gets overwritten by `*_LIBRARY` (very confusing and buggy).

Well, this was just to bug you with this :face_with_head_bandage: :)
I figured it out anyways. 

Anyway, thanks for the clean-up to come :)

**Comment by certik** [comment]
@zerothi Yeah, it bugs me too, it's really messy and not user friendly. CMake has ways to improve. But it still better than the alternatives, overall it allows us to deliver on the task to build on all platforms natively and robustly (once you make CMake finish the configure run successfully...), with the same code. There is very little platform dependent stuff in symengine.

I am going to reopen this issue, as it's a real problem, that hopefully we'll fix once #757 is merged one way or another.

**Comment by zerothi** [comment]
Ok, thanks :)
